### ISSUE TITLE
CircleLoss - loss is always 0 depending on `m` and `gamma`

### ISSUE THREAD

**Issue by mlw214** [KevinMusgrave/pytorch-metric-learning#173]
I've been playing around with CircleLoss, and it seems when you use the paper recommended values of `m=0.25` and `gamma=256`, the loss is always 0.

Here's a simple example of it working with the default `m` and `gamma`:

```python
import torch
from pytorch_metric_learning.losses import CircleLoss

torch.manual_seed(42)

emb = torch.randn(256, 512)
t = torch.randint(0, 10, size=(256,))
lossfn = CircleLoss()

loss = lossfn(emb, t)

print(loss)
# >>> tensor(75.1902)
```

Changing `m` and `gamma` results in a loss of 0.

```python
import torch
from pytorch_metric_learning.losses import CircleLoss

torch.manual_seed(42)

emb = torch.randn(256, 512)
t = torch.randint(0, 10, size=(256,))
lossfn = CircleLoss(m=0.25, gamma=256)

loss = lossfn(emb, t)

print(loss)
# >>> loss = tensor(0.)
```

I played around with other `m` and `gamma` values, and have also seen this behavior. For example, `gamma=128` also doesn't work when `m=0.25`. I have the same issue with the real data I'm using too.

**Comment by KevinMusgrave** [comment]
I think I've fixed it. You can try it out with v0.9.91.dev0:
```
pip install pytorch-metric-learning==0.9.91.dev0
```

**Comment by mlw214** [comment]
Excellent, seems to work at first glace. I'll go ahead and close this for now, and will re-open if necessary.

Thanks for the quick fix, and the excellent library!

**Comment by KevinMusgrave** [comment]
Reopening until its in the "non dev" version
